src/controllers/controllermanager.cpp

```diff
@@ -218,24 +218,24 @@ void ControllerManager::slotSetUpDevices() {
         }
 
         // The filename for this device name.
-        QString presetBaseName = sanitizeString(name);
+        QString deviceName = sanitizeString(name);
 
-        // The first unique filename for this device (appends numbers at the end
-        // if we have already seen a controller by this name on this run of
-        // Mixxx.
-        presetBaseName = firstAvailableFilename(filenames, presetBaseName);
-
-        ControllerPresetPointer pPreset =
-                ControllerPresetFileHandler::loadPreset(
-                    presetBaseName + pController->presetExtension(),
-                    getPresetPaths(m_pConfig));
+        if (m_pConfig->getValueString(ConfigKey("[Controller]", deviceName)) != "1") {
+            continue;
+        }
 
-        if (!loadPreset(pController, pPreset)) {
-            // TODO(XXX) : auto load midi preset here.
+        QString presetFile = m_pConfig->getValueString(
+                ConfigKey("[ControllerPreset]", deviceName));
```

Comment:
Woohoo this finally fixes [using multiple controllers of the same model](https://bugs.launchpad.net/mixxx/+bug/1522507)! I think @mevseme will be happy. :)

Reply:
From here it is a short step to automatically loading mappings instead of making the user select them.